### ISSUE TITLE
Radon unit of measure conversion

### ISSUE THREAD

**Issue by ch0gan** [home-assistant/home-assistant.github.io#31751]
### Feedback

It would be great if this documentation included the steps needed to convert radon Bq/m³ to pCi/L.  Even better if the integration itself offered the option to select a unit of measure.  pCi/L is the standard in the United States.

### URL

https://www.home-assistant.io/integrations/airthings/

### Version

2024.2.5

**Comment by LaStrada** [comment]
Thanks for reporting! Just to clarify, you don't get the correct unit and value with the Airthings cloud integration when setting Home Assistant to use imperial units?

I've not tested this my self, but I cannot see any code that does this conversion from metric to imperial, like we do in the Airthings BLE integration. I'm actually planning on working on something (almost) related later this week, so I can look into this as well.

For reference, here is how it's done in the BLE framework:
https://github.com/Airthings/airthings-ble/blob/main/airthings_ble/parser.py#L584

**Comment by ch0gan** [comment]
So, the value is coming through, but it's coming through in metric (Bq/m³).  Looking at line 471, metric is hard coded in as opposed to looking at the HA settings.  Is it possible to pull the Unit System value from the HA settings?  That would be much better than the work around people are using...  templates and other things I just can't figure out.
Thanks

________________________
Chuck Hogan
***@***.*** 

    On Tuesday, March 5, 2024 at 05:32:44 PM EST, Ståle Storø Hauknes ***@***.***> wrote:  
 
 


Thanks for reporting! Just to clarify, you don't get the correct unit and value with the Airthings cloud integration when setting Home Assistant to use imperial units?

I've not tested this my self, but I cannot see any code that does this conversion from metric to imperial, like we do in the Airthings BLE integration. I'm actually planning on working on something (almost) related later this week, so I can look into this as well.

For reference, here is how it's done in the BLE framework:
https://github.com/Airthings/airthings-ble/blob/main/airthings_ble/parser.py#L584

**Comment by ch0gan** [comment]
I now realize you sent the link to the BLE integration - which I'm not familiar with.  But yes the conversion code there seems like is what's needed in the default HA integration
I'm not a python programmer so I'm not able to propose new code.  What's the process to request changes?
________________________
Chuck Hogan
***@***.*** 

    On Tuesday, March 5, 2024 at 06:26:49 PM EST, Chuck Hogan ***@***.***> wrote:  
 
 So, the value is coming through, but it's coming through in metric (Bq/m³).  Looking at line 471, metric is hard coded in as opposed to looking at the HA settings.  Is it possible to pull the Unit System value from the HA settings?  That would be much better than the work around people are using...  templates and other things I just can't figure out.
Thanks

________________________
Chuck Hogan
***@***.*** 

    On Tuesday, March 5, 2024 at 05:32:44 PM EST, Ståle Storø Hauknes ***@***.***> wrote:  
 
 


Thanks for reporting! Just to clarify, you don't get the correct unit and value with the Airthings cloud integration when setting Home Assistant to use imperial units?

I've not tested this my self, but I cannot see any code that does this conversion from metric to imperial, like we do in the Airthings BLE integration. I'm actually planning on working on something (almost) related later this week, so I can look into this as well.

For reference, here is how it's done in the BLE framework:
https://github.com/Airthings/airthings-ble/blob/main/airthings_ble/parser.py#L584

**Comment by LaStrada** [comment]
I can do a fix later this week

**Comment by ch0gan** [comment]
Great. I’m happy to test when ready. 


Sent from Yahoo Mail for iPhone


On Wednesday, March 6, 2024, 1:59 AM, Ståle Storø Hauknes ***@***.***> wrote:




I can do a fix later this week

**Comment by ch0gan** [comment]
Have you had a chance to look at this?  Looks like it was reported as an issue in January as well, but went stale. https://github.com/home-assistant/core/issues/83015#issue-1470268735

**Comment by LaStrada** [comment]
Yes, looked into this one, but found out there is an architectural discussion around this:
https://github.com/home-assistant/architecture/discussions/732

This could be fixed in the integration, but the best way would be to have a built in unit converter.

**Comment by LaStrada** [comment]
But there are other ways of fixing this. It seems that it's hard to decide on what units to be supported etc for the converter.

So just to be clear, I haven't forgotten this issue ticket! There are some work going on as we speak, and hopefully we'll have a working solution in either the integration itself or in the cloud/framework (pyAirthings).

**Comment by ch0gan** [comment]
Okay to the idea is to fix it in in the core system correct?  Looks like that conversation died two years ago. Is it possible for me to modify the integration code locally?  I have no idea where the python files are stored locally or if they are stored locally. 

I also noticed pressure has the same issue. 
![IMG_5857](https://github.com/home-assistant/home-assistant.io/assets/103752139/17f099e5-5417-4cae-a57a-8dd155801ce7)
![IMG_5856](https://github.com/home-assistant/home-assistant.io/assets/103752139/f05b86e0-0875-4a44-be6b-6c711a72eef7)

**Comment by LaStrada** [comment]
You can change the unit of pressure by:
* Press the entity
* Press the cog in the top right corner
* Change unit

Radon can be fixed by creating a template sensor [here](https://my.home-assistant.io/redirect/config_flow_start?domain=template) and choose "Template a sensor"

<img width="593" alt="image" src="https://github.com/home-assistant/home-assistant.io/assets/270292/c5558d0b-67fd-4028-9fab-7c2e53e69819">

```
{{ states('sensor.airthings_basement_wave_radon') | float / 37 }}
```

Number of decimals can be changed by pressing the newly created entity, pressing the cog, and then changing the "Display precision".

**Comment by ch0gan** [comment]
Thanks.  To be clear, this effectively creates a new "sensor" to be placed on dashboards in place of the default one created by the integration.  Is this correct?

**Comment by LaStrada** [comment]
Yes, you will then have 2 radon sensors (Bq/m³ + pCi/L). Only the sensor provided by the integration will be visible in the Airthings integration (devices & services -> Airthings -> device), the other one will be under helpers (devices & services -> helpers)

**Comment by ch0gan** [comment]
Yes the feature is still wanted but I guess it’s low priority. The goal is a different unit of measurement for radon commonly used in the US. pCi/L


Sent from Yahoo Mail for iPhone


On Wednesday, April 16, 2025, 8:17 PM, github-actions[bot] ***@***.***> wrote:

 
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved.
If this issue is still relevant, please let us know by leaving a comment 👍
This issue has now has been marked as stale and will be closed if no further activity occurs. Thank you for your contributions.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
github-actions[bot] left a comment (home-assistant/home-assistant.io#31751)
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved.
If this issue is still relevant, please let us know by leaving a comment 👍
This issue has now has been marked as stale and will be closed if no further activity occurs. Thank you for your contributions.

**Comment by dalemontoya** [comment]
I would like units in pCi/L please.  Thanks!

**Comment by dmelideo** [comment]
SOLVED, Thank you LaStrada

I wish still active for a built-in solution. I just searched and found this. Appears to possibly work although I have not succeeded.

Edit: I'm not very knowledgeable about HA programming. The solution above worked perfectly. Still need to perfect the new found knowledge of Template Helpers. Created a conversion template for each entity.
Now going to work on a Celsius conversion from Fahrenheit so my devices can provide both values for automation programming.

Thank you LaStrada
